chore(tests): remove `sessionToken` and `sessionTokenWithVerification`

vbudhram opened the pull request:
This PR fixes #306 and will hopefully simplify https://github.com/mozilla/fxa-auth-db-mysql/pull/309.

vbudhram commented on db-server/test/backend/db_tests.js:
```diff
@@ -334,7 +334,7 @@ module.exports = function (config, DB) {
           .then((result) => {
             assert.deepEqual(result, {}, 'Returned an empty object on session token update')
-            return db.sessionToken(sessionTokenData.tokenId)
+            return db.sessionWithDevice(sessionTokenData.tokenId)
```
Updated corresponding tests to use `sessionWithDevice`. This meant that I had to also add checks for `tokenVerificationId` since that value is returned.

vbudhram commented on lib/db/schema/patch-072-073.sql:
```diff
@@ -0,0 +1,53 @@
+
+-- Updates sessionWithDevice procedure to return tokenVerificationCode
+
+CREATE PROCEDURE `sessionWithDevice_12` (
```
Updated this procedure to return the `tokenVerificationCodeHash` since the other session procedures returned this value.

philbooth replied:
Interesting. Do we need to return it?

Presumably if nothing was left that called the other stored procedures, then there's nothing that uses `tokenVerificationCodeHash` otherwise it would be broken. Or is there something broken by the fact that it's not being returned?

No big deal either way, just figure there's no need to return data that isn't being used is all.

vbudhram commented:
@mozilla/fxa-devs r?

rfk commented on db-server/index.js:
```diff
@@ -113,7 +113,6 @@ function createServer(db) {
     })
   )
 
-  api.get('/sessionToken/:id', withIdAndBody(db.sessionToken))
```
From an API standpoint, I wonder if it's worth keeping this endpoint and just changing it to call `db.sessionWithDevice`; it feels a bit odd to have a resource with `put` and `del` but no `get`.

rfk replied:
(IOW, the plain "get session token" method exposed by this API, should just always include the device and verification info)

vbudhram replied:
That seems like a reasonable change to me. We might also be able to replace corresponding [auth-server](https://github.com/mozilla/fxa-auth-server/blob/c805f9c334b99c1162f3389fc5c98d0b5ad602e5/lib/db.js#L535) call to make it less clunky.

philbooth replied:
Agree with this. Would also change the name of the db method to `.sessionToken`, the same way we did in the auth server. (wouldn't change the stored procedure name though, just for the sake of sane versioning if we ever bring back non-device procedure for some reason)

vbudhram commented:
@rfk @philbooth Thanks for feedback! I have updated PR accordingly

* Uses `db.sessionToken` instead of `db.sessionWithDevice`
* Kept the device endpoint until we migrate auth-server to use the regular session endpoint
* Removed migration for tokenVerificationCode since it wasn't used
* Ran tests against auth-server, https://travis-ci.org/mozilla/fxa-auth-server/builds/344868321?utm_source=github_status&utm_medium=notification

@mozilla/fxa-devs r?

philbooth commented on docs/DB_API.md:
```diff
@@ -626,7 +619,7 @@ Parameters:
     * createdAt
 
-## .sessionWithDevice(tokenId) ##
+## .sessionToken(tokenId) ##
```
Could we delete this section entirely if we added this bit and the associated `d.*` fields to the preceding section:

```
`d.*` for a field from `devices` and
```

?
